### PR TITLE
Try to get the value first to avoid creating a lambda

### PR DESCRIPTION
**Pull request by guidomedina**
Try to get the value first which is most likely to be cached to avoid creating a lambda.

**Similar to:**
https://github.com/quickfix-j/quickfixj/blob/master/quickfixj-core/src/main/java/org/quickfixj/SimpleCache.java

I was going to submit this PR yesterday but I was so busy that got distracted and forgot.

### CODE REVIEW

**Comment by guidomedina** [comment]
I was going to submit this PR yesterday but I was so busy that got distracted and forgot.

**Comment by svendiedrichsen** [comment]
@guidomedina I mad some comments in the code.

**Comment by guidomedina** [comment]
For some reason the same thing that happened last time is happening again; when I look at the commit and files changed it doesn't show your comment/review, I have seen reviews before of my PRs in other repos but never had this issue before.

In fact I even get e-mails with the comments at the code which I haven't got, very strange.

**Comment by guidomedina** [comment]
@svendiedrichsen I can't see your comments, I don't know what to do, I even tried from an incognito session and your comments are not there.

**Comment by svendiedrichsen** [comment]
@guidomedina If you are logged in you can see them right in this conversation or when you are looking at the 'Files changed' tab.

File: src/main/java/de/jollyday/util/Cache.java [hunk 1]
```diff
@@ -37,7 +37,10 @@
 	 * @return the eventually cached value
 	 */
 	public VALUE get(ValueHandler<VALUE> valueHandler) {
-		return cachingMap.computeIfAbsent(valueHandler.getKey(), k -> valueHandler.createValue());
+		final String key = valueHandler.getKey();
+		// Try to first get the value which is most likely cached to avoid creating a lambda.
+		final VALUE value = cachingMap.get(key);
+		return value != null ? value : cachingMap.computeIfAbsent(key, k -> valueHandler.createValue());
```
Comment (guidomedina) [inline src/main/java/de/jollyday/util/Cache.java]: @svendiedrichsen can you see this comment? I know where comments are made in the code, but trust me when I tell you that I cannot see your comments.

**Comment by svendiedrichsen** [comment]
Comments:
1) What is the reason for calling getDeclaredConstructor Method?
2) Why check cache first? Isn't this the exact purpose of the computIfAbsent Method?

**Comment by guidomedina** [comment]
- In Java 9+ `newInstance()` is deprecated, `newInstance()` will get the declared constructor without parameters and create a new instance anyway (I can undo this part if you want, I'm running IntelliJ with Java 9 and it shown as deprecated when I analysed the code)
- `computeIfAbsent` will lock that hash bucket internally and create a new lambda because the `ValueHandler` function is not static but dynamic, even though Java is very effective at caching lambdas it is also a mechanism that it is easily exploited, for our case it is most likely that the value was computed before and hence avoid a lambda creation.

If the function were static and not a lambda dynamically created the difference would be very minimal; for caches with high reading pattern vs write calling first get is more effective.

For example, in [SimpleCache](https://github.com/quickfix-j/quickfixj/blob/master/quickfixj-core/src/main/java/org/quickfixj/SimpleCache.java) there is a pre-defined loading function for the whole cache and not one created per key.

**Comment by svendiedrichsen** [comment]
Ok, sounds good to me then. Thanks for the work.

**Pull request merged**